torch/csrc/autograd/saved_variable.cpp

```diff
@@ -154,6 +194,9 @@ Variable SavedVariable::unpack(std::shared_ptr<Node> saved_for) const {
 }
 
 void SavedVariable::register_hooks(std::unique_ptr<SavedVariableHooks>&& hooks) {
+  TORCH_CHECK(!hooks_,
```

Comment:
Would be good to see a test for this also!

Reply:
This check was added with the previous PR so it was tested back then (https://github.com/pytorch/pytorch/pull/60663/files#diff-964311d1ab760238d76cb29a0e25e64f2c18cb3dbb076e10d6337c59d6dd718aR4816)

Reply:
Oh whoops, sorry